pkgs/games/dwarf-fortress/game.nix

```
@@ -57,25 +72,42 @@ let
     if hasAttr dfPlatform game
     then getAttr dfPlatform game
     else throw "Unsupported dfPlatform: ${dfPlatform}";
-
+  exe = if stdenv.isLinux then
+    if baseVersion >= 50 then "dwarfort" else "libs/Dwarf_Fortress"
+  else
+    "dwarfort.exe";
 in
 
 stdenv.mkDerivation {
   pname = "dwarf-fortress";
   version = dfVersion;
 
   src = fetchurl {
-    url = "https://www.bay12games.com/dwarves/df_${baseVersion}_${patchVersion}_${dfPlatform}.tar.bz2";
+    url = "https://www.bay12games.com/dwarves/df_${toString baseVersion}_${toString patchVersion}_${dfPlatform}.tar.bz2";
     inherit sha256;
   };
 
+  sourceRoot = ".";
+
+  postUnpack = optionalString stdenv.isLinux ''
+    if [ -d df_linux ]; then
+      mv df_linux/* .
+    fi
+  '' + optionalString stdenv.isDarwin ''
+    if [ -d df_osx ]; then
+      mv df_osx/* .
+    fi
+  '';
+
+  nativeBuildInputs = lib.optionals isV50 [ autoPatchelfHook ];
+  buildInputs = lib.optionals isV50 [ SDL2 SDL2_image SDL2_mixer stdenv.cc.cc.lib ];
+
   installPhase = ''
     mkdir -p $out
     cp -r * $out
-    rm $out/libs/lib*
+    ${lib.optionalString (!isV50) "rm -f $out/libs/*.so"}
```

```suggestion
  '' + lib.optionalString (!isV50) ''
    rm -f $out/libs/*.so
  '' + ''
```
we can save ourselves string interpolation strugle